addons/zylann.hterrain/native/src/quad_tree_lod.h

```diff
@@ -3,120 +3,125 @@
 
 #include <Godot.hpp>
 #include <FuncRef.hpp>
+#include <CanvasItem.hpp>
+
+#include <vector>
+
 
 namespace godot {
 
 class QuadTreeLod : public Reference {
     GODOT_CLASS(QuadTreeLod, Reference)
 
+    static const unsigned int NO_CHILDREN = -1;
+    static const unsigned int ROOT = -1;
+
     class Quad {
+
     private:
-        Quad **children = nullptr;
-        bool _has_children = false;
+        Variant _data;          // Type is Object: HTerrainChunk.gd
 
     public:
+        unsigned int first_child = NO_CHILDREN;
         int origin_x = 0;
         int origin_y = 0;
-        Variant data;          // Type is HTerrainChunk.gd
 
         Quad() {
-            //Always allocate memory for child container, but not children
-            children = new Quad* [4];
-            for (int i = 0; i < 4; i++)
-                children[i] = nullptr;
+            init();
         }
 
         ~Quad() {
-            clear_children();
-            delete[] children;
         }
 
-        void clear() {
-            clear_children();
-            data = Variant();
+        inline void init() {
+            first_child = NO_CHILDREN;
+            origin_x = 0;
+            origin_y = 0;
+            clear_data();
         }
 
-        void add_children() {
-            if (_has_children)
-                Godot::print("Error: Quad::add_children(): already has children");
-                return;
+        inline void clear_data() {
+            _data = Variant();
```

Comment:
Does this adequately remove our reference to the underlying object? I wasn't sure about it looking at the variant code. It seemed to me that I needed to run Variant::clear() to remove the reference, but that's not available in godot-cpp. The only way to get it is to call the destructor. But then I'd have to use pointers and new so I could delete _data, right?

Reply:
I assume replacing with an empty variant is similar to assigning to `null` in GDScript.

Reply:
Ok, I won't worry about it then.